x/transfermiddleware/keeper/keeper.go

```diff
@@ -92,7 +92,7 @@ func (keeper Keeper) AddParachainIBCInfo(ctx sdk.Context, ibcDenom, channelID, n
 func (keeper Keeper) AddParachainIBCInfoToRemoveList(ctx sdk.Context, nativeDenom string) (time.Time, error) {
 	params := keeper.GetParams(ctx)
 	store := ctx.KVStore(keeper.storeKey)
-	if store.Has(types.GetKeyParachainIBCTokenInfoByNativeDenom(nativeDenom)) {
```

Comment:
Old logic isn't exact. We need to check key is exist and if it isn't throw an  error